### PR TITLE
rc/wp - 6pm-2 Add Swagger support to enable easier development of backend apis

### PR DESCRIPTION
**Pull request by Wilson-Flan**
As a developer, I can use swagger-ui to access backend endpoint easily so that it is easier to do backend development.

### CODE REVIEW

File: src/main/java/edu/ucsb/ucsbcslas/controllers/ReactController.java [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 @Controller
 public class ReactController {
-  @RequestMapping(value = {"/", "/{x:[\\w\\-]+}", "/{x:^(?!api$).*$}/**/{y:[\\w\\-]+}"})
+  @RequestMapping(value = {"/", "/{x:^(?!swagger-ui$)[\\w\\-]+}", "/{x:^(?!api$).*$}/**/{y:[\\w\\-]+}"})
```
Comment (rowantran) [inline src/main/java/edu/ucsb/ucsbcslas/controllers/ReactController.java:8]: does anybody know what this regex is doing

**Review by rowantran — APPROVED** [review]
LGTM. the regex is like wizardry to me but code seems to work

File: src/main/java/edu/ucsb/ucsbcslas/controllers/ReactController.java (hunk 1 above)
Comment (pconrad) [reply src/main/java/edu/ucsb/ucsbcslas/controllers/ReactController.java:8]: Honestly... not completely!   

So, I do know *some* of what it is doing.  It is trying to say something like: "use this controller for the home page (`/`) and every URL *except* those that start with /api and those that start with /swagger-ui.

But there is also some deep black magic going on that I don't fully understand.  I tried simplifying these in ways that I *thought* should work, to make them less .. weird... and more readable.  I was unsuccessful.

Comment (pconrad) [reply src/main/java/edu/ucsb/ucsbcslas/controllers/ReactController.java:8]: To clarify: the original lines of code here were written by @scottpchow23 (who got them from idk where), and then I hacked at them to get swagger to work.

**Review by pconrad — APPROVED** [review]
LGTM.... pending testing on the qa site.

I'm approving a merge in spite of the codecov stats showing a small dip in code coverage.

**Pull request merged**